src/main/java/com/autotune/analyzer/recommendations/model/CostBasedRecommendationModel.java

```diff
@@ -216,9 +239,15 @@ private static double calculateMemoryUsage(IntervalResults intervalResults) {
         if (0 < numPods) {
             memUsage = (memUsageSum / numPods);
         }
-        memUsage = Math.max(memUsage, memUsageMax);
+        memUsageMax = Math.max(memUsage, memUsageMax);
+        memUsageMin = Collections.min(Arrays.asList(memUsage, memUsageMax, memUsageMin));
+
+        jsonObject.put(KruizeConstants.JSONKeys.MAX, memUsageMax);
+        if (memUsageMin > 0.0)
```

Comment:
Should ignore the variable which is zero and continue to find minimum value from the others. It can be completely avoided only if all the variables(memUsage, memUsageMax, memUsageMin) are zero.